pkg/operator/ceph/csi/csi.go

```diff
@@ -179,6 +179,10 @@ func (r *ReconcileCSI) setParams(ver *version.Info) error {
 	if strings.EqualFold(k8sutil.GetValue(r.opConfig.Parameters, "CSI_ENABLE_CSIADDONS", "false"), "true") {
 		CSIParam.EnableCSIAddonsSideCar = true
 	}
+	_, err = r.context.ApiExtensionsClient.ApiextensionsV1().CustomResourceDefinitions().Get(r.opManagerContext, "networkfences.csiaddons.openshift.io", metav1.GetOptions{})
+	if err == nil {
+		CSIParam.EnableCSIAddonsSideCar = true
+	}
```

Comment:
```suggestion
CSIParam.EnableCSIAddonsSideCar = false
	_, err = r.context.ApiExtensionsClient.ApiextensionsV1().CustomResourceDefinitions().Get(r.opManagerContext, "networkfences.csiaddons.openshift.io", metav1.GetOptions{})
	if err == nil {
		CSIParam.EnableCSIAddonsSideCar = true
	}
if strings.EqualFold(k8sutil.GetValue(r.opConfig.Parameters, "CSI_ENABLE_CSIADDONS", ""), "false") {
		CSIParam.EnableCSIAddonsSideCar = false
	}
```

i think above might be the good as we still allow users to disable csi_addons if they want

Reply:
Agreed, this looks good as long as the user has the ability to disable the sidecar if desired.